x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -328,44 +334,55 @@ export const statusQueryParamsSchema = schema.object({
   check: schema.oneOf([schema.literal('all'), schema.literal('init')], { defaultValue: 'all' }),
 });
 
-export const defineGetCspStatusRoute = (router: CspRouter): void =>
-  router.get(
-    {
+export const defineGetCspStatusRoute = (
+  router: CspRouter
+): VersionedRoute<'get', CspRequestHandlerContext> =>
+  router.versioned
+    .get({
+      access: 'internal',
       path: STATUS_ROUTE_PATH,
-      validate: { query: statusQueryParamsSchema },
       options: {
         tags: ['access:cloud-security-posture-read'],
       },
-    },
-    async (context, request, response) => {
-      const cspContext = await context.csp;
-      try {
-        if (request.query.check === 'init') {
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            query: statusQueryParamsSchema,
```

Comment:
should the request and the implementation should be versioned as well?

I'm asking because from a look on the code here it is not clear what is expected when adding a new version and eventually it will require additional refactoring.
So I recommend demonstrating how a newer version should be applied

cc: @opauloh

Reply:
> should the request and the implementation should be versioned as well?

Yes the request is versioned as well, please go over `x-pack/plugins/cloud_security_posture/public/common/api/use_setup_status_api.ts` in this PR, or click [here](https://github.com/elastic/kibana/pull/159548/files#diff-8a1665570ffb46e36bfc31d1bd12d64a0e36bdcc9da14953590a72e660f1d7b9R17)

> So I recommend demonstrating how a newer version should be applied

The process of adding a new version is thoroughly [documented](https://docs.elastic.dev/kibana-dev-docs/versioning-http-apis) in the documentation.

<details>
  <summary>Example</summary>
  
```typescript
  router.versioned.
    .post({
      access: 'public', // This endpoint is intended for a public audience
      path: '/api/my-app/foo/{id?}',
      options: { timeout: { payload: 60000 } },
    })
    .addVersion(
      {
        version: '2023-10-31', // The public version of this API
        validate: {
          request: {
            query: schema.object({
              name: schema.maybe(schema.string({ minLength: 2, maxLength: 50 })),
            }),
            params: schema.object({
              id: schema.maybe(schema.string({ minLength: 10, maxLength: 13 })),
            }),
            body: schema.object({ foo: schema.string() }),
          },
          response: {
            200: { // In development environments, this validation will run against 200 responses
              body: schema.object({ foo: schema.string() }),
            },
          },
        },
      },
      async (ctx, req, res) => {
        await ctx.fooService.create(req.body.foo, req.params.id, req.query.name);
        return res.ok({ body: { foo: req.body.foo } });
      }
    )
    // BREAKING CHANGE: { foo: string } => { fooString: string } in response body
    .addVersion(
      {
        version: '2024-10-31',
        validate: {
          request: {
            query: schema.object({
              name: schema.maybe(schema.string({ minLength: 2, maxLength: 50 })),
            }),
            params: schema.object({
              id: schema.maybe(schema.string({ minLength: 10, maxLength: 13 })),
            }),
            body: schema.object({ fooString: schema.string() }),
          },
          response: {
            200: {
              body: schema.object({ fooName: schema.string() }),
            },
          },
        },
      },
      async (ctx, req, res) => {
        await ctx.fooService.create(req.body.fooString, req.params.id, req.query.name);
        return res.ok({ body: { fooName: req.body.fooString } });
      }
    )
    // BREAKING CHANGES: Enforce min/max length on fooString
    .addVersion(
      {
        version: '2025-03-01',
        validate: {
          request: {
            query: schema.object({
              name: schema.maybe(schema.string({ minLength: 2, maxLength: 50 })),
            }),
            params: schema.object({
              id: schema.maybe(schema.string({ minLength: 10, maxLength: 13 })),
            }),
            body: schema.object({ fooString: schema.string({ minLength: 0, maxLength: 1000 }) }),
          },
          response: {
            200: {
              body: schema.object({ fooName: schema.string() }),
            },
          },
        },
      },
      async (ctx, req, res) => {
        await ctx.fooService.create(req.body.fooString, req.params.id, req.query.name);
        return res.ok({ body: { fooName: req.body.fooString } });
      }
  ```
</details>

I don't sure I understand, do you want me to add code in comment that explains how to add a new version?

Reply:
Thanks for the response, so I think in this opportunity we should refactor the code
That the versioned callback translates the call to the BL

Reply:
Resolving the comment - we've discussed that you'll introduce new impl. details and refactor  all the versioned endpoint

cc: @JordanSh